pkg/timelock/timelock.go

```diff
@@ -273,6 +336,31 @@ func (tw *Worker) retrieveHistoricalLogs(ctx context.Context) (<-chan struct{},
 	return done, logCh, nil
 }
 
+func (tw *Worker) fetchAndDispatchLogs(ctx context.Context, logCh chan types.Log, lastBlock *big.Int) *big.Int {
```

Comment:
this is partially adapted from the existing `retrieveHistoricalLogs` to try to maintain the consumer code the same